coderd/database/modelmethods.go

```diff
@@ -124,6 +124,15 @@ func (t Template) DeepCopy() Template {
 	return cpy
 }
 
+// AutostartAllowedDays returns the inverse of 'AutostartBlockDaysOfWeek'.
+// It is more useful to have the days that are allowed to autostart from a UX
+// POV. The database prefers the 0 value being 'all days allowed'.
+func (t Template) AutostartAllowedDays() uint8 {
+	// Just flip the binary 0s to 1s and vice versa.
+	// There is an extra day with the 8th bit that needs to be zeroed.
+	return ^uint8(t.AutostartBlockDaysOfWeek) & 0b01111111
```

Comment:
It's even more strange that we store a blocklist and then expose this function to flip it 🤔

Reply:
From the UX POV, I think the deny list is more difficult to understand, so I flip it when returning it from the BE.

Reply:
Isn't it simpler to just store an allowlist though e.g. `AutostartAllowedDaysOfWeek`?

Reply:
Consumers of the API will still have to submit a denylist, right?

Reply:
No, because we gatekeep all schedule changes by the AGPL and Enterprise interfaces, the inversion happens right before we do the db query.

https://github.com/coder/coder/blob/25c3b1ba1c28408190b2773781c8f87a04accff4/enterprise/coderd/schedule/template.go#L150-L150


Consumers of the `database.Template` struct do see the denylist, and should use this model method to get the allowlist: https://github.com/coder/coder/blob/25c3b1ba1c28408190b2773781c8f87a04accff4/coderd/database/modelmethods.go#L130-L130

There's no easy way to encapsulate this to hide the underlying implementation other than doing something overkill like making a custom uint8 type with a custom postgres scanner. Feels overkill though.

Reply:
And another comment is I cannot write that function in the schedule package because that imports the database package. So I just wrote the binary flip :man_shrugging:. Should be the only spot it's needed though. Most of the binary logic comes from that schedule package